source/_integrations/xiaomi_miio.markdown

```diff
@@ -64,6 +64,13 @@ you used in the Xiaomi Home App (where you initially setup the device). There ar
 [this page](https://www.openhab.org/addons/bindings/miio/#country-servers) for
 the server to use for each country.
 
+_Note: Roborock vacuums must be set up using the Mi Home app and not the Roborock app to use this integration. Additionally, if the following devices and Home Assistant are on different subnets/VLANs, [additional configuration](https://python-miio.readthedocs.io/en/latest/troubleshooting.html#discover-devices-across-subnets) on your router is required._
```

Comment:
Maybe split this into two separate notes?
1. The devices need to be set up using the Mi Home app, and not with vendor-specific apps
2. Some devices need to be directly connected / in the same subnet as homeassistant (link to python-miio docs as you already had)

Reply:
That sounds good. I just made some minor tweaks to the wording. Thoughts on this? Doesn’t matter to me either way, just trying to make it concise and clear as possible.

Notes:
- Devices need to be set up using the Mi Home app and not vendor-specific apps (e.g. Roborock).
- Some devices need to be on the same subnet/VLAN as Home Assistant or [additional configuration](https://python-miio.readthedocs.io/en/latest/troubleshooting.html#discover-devices-across-subnets) will be required.

Reply:
our docs do not cover subnet configuration. Advanced networking is considered out of scope for our documentation; if one uses such network setups, one should be able to deal with that.

Reply:
Whoops, sorry I completely missed that section of the documentation standard. I’ll update that. As for mentioning subnet/VLANs, I would argue this is a bit different since it’s fairly unique. Most integrations I’ve used with the devices on a separate subnet/VLAN work just fine and I would completely agree if someone is doing that, they will also understand firewall rules need to be configured, etc. (which isn’t mentioned). However, this is one of the only devices I’ve seen that requires IP masquerading for the integration to work. Having just gone through this process, I think having a simple note with a link to the python-miio documents will save someone a lot of time/headache (which was me yesterday, lol). 😄

Reply:
That sounds good to me, the wording I used was just to give an example about separating the issues :-)

Reply:
>  I would argue this is a bit different since it’s fairly unique to this integration

Sorry, it (unfortunately) isn't.

Reply:
No worries, I’ll just remove the second note. Just seems a bit odd we have an integration where some devices will work just fine on a separate subnet/VLAN, but a handful will not, and putting a note to point that out is beyond the scope of the integration documentation. 🤷‍♂️ But, the note about the app will definitely help - that wasted about an hour of my day trying to figure out why the cloud credentials were not being accepted. 😆

Reply:
It is hard to draw lines between what to document or not on advanced networking. For example, it could start to differ based on the network equipment used or the way one has set up their networking.

The complicity added, is that we would start to accept all kinds of instructions for all kinds of network equipment or idea to our documentation; while in the end, it covers only an edge case that is used less common and only by advanced users that choose to take a more complex route. From a maintenance and correctness perspective, we don't add these parts to the documentation for that reason.

Maybe we could make the note smaller, more generic: "If you are using a more complex network setup; the following documentation might help you" or something. Without going into specific details.

Reply:
I completely understand - I wouldn’t want to put instructions on how to set up 1:1 NAT or IP Masquerading in the documents either, nor was I suggesting that. I was just thinking a small note that points out if you’re using multiple subnets or VLANs, to reference the python-miio troubleshooting page because I know I’m not the [only person](https://community.home-assistant.io/t/xiaomis-vacuum-roborock-s7-fails-to-connect/362715/2) (or last) to have this issue. 

I appreciate you guys trying to keep the documentation concise and clean. I can only imagine how chaotic it would be if you guys weren’t managing it. 😄

I can add a note back if you want, something along the lines of “If you are using a more complex network setup (e.g. VLANs), reference the following documentation for additional information.”

Otherwise, happy to add just the note about the app.

Reply:
Sounds good 👍